Source/Microsoft.Teams.Apps.FAQPlusPlus/Bots/FaqPlusPlusBot.cs

```diff
@@ -1402,15 +1413,26 @@ private async Task<bool> IsMemberOfSmeTeamAsync(ITurnContext<IInvokeActivity> tu
         /// Get the reply to a question asked by end user.
         /// </summary>
         /// <param name="turnContext">Context object containing information cached for a single turn of conversation with a user.</param>
-        /// <param name="text">Text message.</param>
+        /// <param name="message">Text message.</param>
         /// <returns>A task that represents the work queued to execute.</returns>
         private async Task GetQuestionAnswerReplyAsync(
             ITurnContext<IMessageActivity> turnContext,
-            string text)
+            IMessageActivity message)
         {
+            string text = message.Text?.ToLower()?.Trim() ?? string.Empty;
+
             try
             {
-                var queryResult = await this.qnaServiceProvider.GenerateAnswerAsync(question: text, isTestKnowledgeBase: false).ConfigureAwait(false);
+                var queryResult = new QnASearchResultList();
+
+                ResponseCardPayload payload = new ResponseCardPayload();
+
+                if (!string.IsNullOrEmpty(message.ReplyToId) && (message.Value != null) && ((JObject)message.Value).HasValues)
```

Comment:
Is the check for HasValues necessary? I think we can remove this. (@abbodh ?)

Reply:
Removed the HasValues check. Added that for extra validation.